nautobot_device_lifecycle_mgmt/migrations/0021_migrate_to_core_software_models.py

```diff
@@ -0,0 +1,264 @@
+        core_software_image = CoreSoftwareImage(
+            id=dlm_software_image.id,
+            software_version=CoreSoftwareVersion.objects.get(id=dlm_software_image.software.id),
+            image_file_name=dlm_software_image.image_file_name,
+            image_file_checksum=dlm_software_image.image_file_checksum,
+            hashing_algorithm=_migrate_hashing_algorithm(dlm_software_image.hashing_algorithm),
+            download_url=dlm_software_image.download_url,
+            status=status_active,
+            _custom_field_data=dlm_software_image._custom_field_data or {},
+        )
+        core_software_image.save()
+        core_software_image.device_types.set(dlm_software_image.device_types.all())
+
+        # Work around created field's auto_now_add behavior
+        CoreSoftwareImage.objects.filter(id=core_software_image.id).update(created=dlm_software_image.created)
+
+        # Map the DLM object_tags to devices and set the Device.software_image_files m2m field
+        device_pks = (
+            TaggedItem.objects.filter(tag__in=dlm_software_image.object_tags.all(), content_type=device_ct)
+            .values_list("object_id")
+            .distinct()
+        )
+        for device in Device.objects.filter(pk__in=device_pks):
+            device.software_image_files.add(core_software_image)
+
+        # TODO: Map the DLM object_tags to inventory items and set the InventoryItem.software_image_files m2m field
+        # TODO: Map the DLM object_tags to virtual machines and set the VirtualMachine.software_image_files m2m field
+
+        core_software_image.refresh_from_db()
+
+        # make tag manager available in migration for nautobot.core.models.utils.serialize_object
+        # https://github.com/jazzband/django-taggit/issues/101
+        # https://github.com/jazzband/django-taggit/issues/454
+        core_software_image.tags = TagsManager(
+            through=extras_models.TaggedItem,
+            model=CoreSoftwareImage,
+            instance=core_software_image,
+            prefetch_cache_name="tags",
+        )
+
+        # Create an object change to document migration
+        ObjectChange.objects.create(
+            action=extras_choices.ObjectChangeActionChoices.ACTION_UPDATE,
+            change_context=extras_choices.ObjectChangeEventContextChoices.CONTEXT_ORM,
+            change_context_detail="Migrated from Nautobot App Device Lifecycle Management",
+            changed_object_id=core_software_image.id,
+            changed_object_type=core_software_image_ct,
+            object_data=serialize_object(core_software_image),
+            object_repr=f"{core_software_image.software_version.platform.name} - "
+            f"{core_software_image.software_version.version} - "
+            f"{core_software_image.image_file_name}"[:CHANGELOG_MAX_OBJECT_REPR],
+            request_id=uuid.uuid4(),
+        )
+
+
+def _migrate_content_types(apps, old_ct, new_ct):
```

Comment:
Wonder if it would make sense to make this function a part of core instead - seems pretty generic to any case where we're migrating models from another app?